src/openrct2/scenario/ScenarioRepository.cpp

```diff
@@ -285,6 +285,7 @@ class ScenarioRepository final : public IScenarioRepository
         Path::Append(mpdatPath, sizeof(mpdatPath), "mp.dat");
 
         String::Set(sc21Path, sizeof(sc21Path), scenarioDir.c_str());
+        platform_ensure_directory_exists(sc21Path); //Ensure the scenario directory exists first
```

Comment:
Comments should go above the line they're referring to (see [style guide here](https://github.com/OpenRCT2/OpenRCT2/wiki/Coding-Style)).